src/pygeoml200/calibration.py

```diff
@@ -50,10 +50,11 @@ def place_calibration_system(b: core.InstrumentationData) -> None:
     sis_cfg = b.runtime_config.get("sis", {})
 
     for i, tube in b.special_metadata.calibration.items():
-        idx = int(i) - 1
+        i_int = int(i)
+        idx = i_int - 1
         if tube is None:
             continue
-        tube_cfg = sis_cfg.get(i, {}) or {}
+        tube_cfg = sis_cfg.get(i_int, {}) or {}
```

Comment:
now the two key schemes are inconstistent. The inner (per-sis) dict allows both str/int keys, the outer one only supports int keys?